src/lib/rnp.cpp

```diff
@@ -5675,33 +5670,42 @@ static rnp_result_t
 rnp_key_get_signature_count_for_uid(pgp_key_t *key, size_t *count, uint32_t uid)
 {
     *count = 0;
-    for (size_t i = 0; i < pgp_key_get_subsig_count(key); i++) {
-        if (pgp_key_get_subsig(key, i)->uid == uid) {
+    for (size_t i = 0; i < key->sig_count(); i++) {
+        if (key->get_sig(i).uid == uid) {
             (*count)++;
         }
     }
     return RNP_SUCCESS;
 }
 
+static rnp_result_t
+rnp_key_return_signature(rnp_ffi_t               ffi,
+                         pgp_key_t *             key,
+                         pgp_subsig_t &          subsig,
```

Comment:
It's a bit confusing.
the new object captures the pointer `key` and puts it into the newly created object, 
it also captures the pointer `&subsig`
So let's have parameter of type `pgp_subsig_t *`

Reply:
Thanks, fixed.